src/LFHCAL_geo.cpp

```diff
@@ -0,0 +1,1125 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2023 Friederike Bock
+
+//==========================================================================
+//  Implementation of longitudinally separated forward calorimeter
+//--------------------------------------------------------------------------
+//  Author: Friederike Bock (ORNL)
+//==========================================================================
+
+#include "DD4hep/DetFactoryHelper.h"
+#include "DD4hep/Printout.h"
+#include "DD4hep/Shapes.h"
+#include "DDRec/DetectorData.h"
+#include "DDRec/Surface.h"
+#include <XML/Helper.h>
+#include "XML/Layering.h"
+#include "XML/Utilities.h"
+using namespace dd4hep;
+
+struct moduleParamsStrct{
+  moduleParamsStrct(): mod_BIwidth(0.), mod_BIheight(0.), mod_SWThick(0.),   mod_TWThick(0.), mod_FWThick (0.),
+                      mod_BWThick(0.), mod_width(0.), mod_height(0.), mod_notchWidthAbsA(0.), mod_notchWidthAbsB(0.), mod_notchWidthAbsC (0.),
+                      mod_notchWidthScin(0.), mod_notchDepth(0.), mod_sepDepth(0.),  mod_visStr(""), mod_regStr(""), mod_limStr("")
+                      {}
+  moduleParamsStrct(   double BIwidth, double BIheight, double SWThick, double TWThick, double FWThick, double BWThick, double width, double height,
+                       double notchWidthAbsA, double notchWidthAbsB, double notchWidthAbsC, double notchWidthScin, double notchDepth, double sepDepth,
+                       std::string visStr, std::string regStr, std::string limStr){
+      mod_BIwidth       = BIwidth;
+      mod_BIheight      = BIheight;
+      mod_SWThick       = SWThick;
+      mod_TWThick       = TWThick;
+      mod_FWThick       = FWThick;
+      mod_BWThick       = BWThick;
+      mod_width         = width;
+      mod_height        = height;
+      mod_notchWidthAbsA  = notchWidthAbsA;
+      mod_notchWidthAbsB  = notchWidthAbsB;
+      mod_notchWidthAbsC  = notchWidthAbsC;
+      mod_notchWidthScin  = notchWidthScin;
+      mod_notchDepth      = notchDepth;
+      mod_sepDepth        = sepDepth;
+      mod_visStr          = visStr;
+      mod_regStr          = regStr;
+      mod_limStr          = limStr;
+  }
+  double      mod_BIwidth;
+  double      mod_BIheight;
+  double      mod_SWThick;
+  double      mod_TWThick;
+  double      mod_FWThick;
+  double      mod_BWThick;
+  double      mod_width;
+  double      mod_height;
+  double      mod_notchWidthAbsA;
+  double      mod_notchWidthAbsB;
+  double      mod_notchWidthAbsC;
+  double      mod_notchWidthScin;
+  double      mod_notchDepth;
+  double      mod_sepDepth;
+  std::string mod_visStr;
+  std::string mod_regStr;
+  std::string mod_limStr;
```

Comment:
For clarity and to make all fields are always initialized, can we add the default values with `{}` initializes here and get rid of the default constructor?

Reply:
prefer to keep as is.

Reply:
I now initialized also all fields, but I would like to keep the default constructor, which I think is good practise.

Reply:
Sure, that's fine. You probably know this, but https://en.cppreference.com/w/cpp/language/rule_of_three (and rules of zero and five included therein; naturally rule of zero is least work).